pkg/cmd/experimental/buildchain/buildchain.go

```diff
@@ -0,0 +1,164 @@
+package buildchain
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	// "code.google.com/p/gographviz"
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	cmdutil "github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd/util"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+
+	buildapi "github.com/openshift/origin/pkg/build/api"
+	"github.com/openshift/origin/pkg/client"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	_ "github.com/openshift/origin/pkg/image/api"
+	val "github.com/openshift/origin/pkg/image/api/validation"
+)
+
+// node is a representation of a node inside a tree
+type node struct {
+	name     string
+	tags     []string
+	children []*node
+}
+
+// String helps in pretty-printing a tree
+func (root *node) String() string {
+	tree := ""
+	tree += root.name
+
+	for _, n := range root.children {
+		tree += "(" + n.String() + ")"
+	}
+
+	return tree
+}
+
+const longDescription = `Experimental command
+
+Output build dependencies of a specific image repository.
+
+Examples:
+
+    $ openshift ex build-chain [image-repository]
+`
+
+func NewCmdBuildChain(f *clientcmd.Factory, parentName, name string) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   fmt.Sprintf("%s [image-repository]", name),
+		Short: "Output build dependencies of a specific image repository",
+		Long:  longDescription,
+		Run: func(cmd *cobra.Command, args []string) {
+			imageRepoName := cmdutil.GetFlagString(cmd, "imagerepo")
+			if imageRepoName == "" {
+				usageError(cmd, "Must pass an image repository name with the --imagerepo (-i) flag")
+			}
+
+			c, _, err := f.Clients(cmd)
+			checkErr(err)
+
+			namespace, err := f.DefaultNamespace(cmd)
+			checkErr(err)
+
+			// Validate the specified image repository
+			if !isValid(imageRepoName, c, namespace) {
+				checkErr(fmt.Errorf("invalid image repository: %s", imageRepoName))
+			}
+
+			// Get all build configurations and look for those
+			// that reference the specified image repository
+			buildConfigList, err := c.BuildConfigs(namespace).List(labels.Everything(), labels.Everything())
+			checkErr(err)
+
+			root := findRepoDeps(imageRepoName, buildConfigList.Items)
+			fmt.Println(root)
+			// imgRoot := repoToImage(c, namespace, imageRepoName, root)
+		},
+	}
+
+	cmd.Flags().StringP("imagerepo", "i", "", "Image repository for which to output build dependencies")
+	return cmd
+}
+
+// isValid checks if the given image repository is valid
+func isValid(name string, c *client.Client, namespace string) bool {
+	// TODO: Validate image repository via buildConfigs?
```

Comment:
Since we probably haven't run any buildConfigs, these image repositories do not exist, so anything that goes here will be baptized as invalid. So I think we should check for image repositories in buildConfigs right? 

(cc: @bparees @csrwng )